dash/dash.py

```diff
@@ -676,6 +679,18 @@ def _generate_meta_html(self):
 
     # Serve the JS bundles for each package
     def serve_component_suites(self, package_name, path_in_package_dist):
+        # Check if the resource has a fingerprint
+        res = re.match(
+            r'^(.*)[.]v\d+_\d+_\d+(-\w+[.]?\d+)?m\d{10}([.]js)$',
+            path_in_package_dist,
+        )
+
+        fingerprint = res is not None
+        # Resolve real resource name from fingerprinted resource path
+        path_in_package_dist = (
+            res[1] + res[3] if res is not None else path_in_package_dist
+        )
```

Comment:
If the resource was requested with the correct pattern (from dash filling the html page or from async resources override) - use cache-control, otherwise use Etag